src/main/java/io/aiven/kafka/connect/transforms/TombstoneHandler.java

```
@@ -0,0 +1,73 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.transforms;
+
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.Transformation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TombstoneHandler<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TombstoneHandler.class);
+
+    private TombstoneHandlerConfig tombstoneHandlerConfig;
+
+    @Override
+    public ConfigDef config() {
+        return TombstoneHandlerConfig.config();
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        this.tombstoneHandlerConfig = new TombstoneHandlerConfig(configs);
+    }
+
+    @Override
+    public R apply(final R record) {
+        if (record.valueSchema() == null || record.value() == null) {
+            behaveOnNull(record);
+        }
+        return record;
+    }
+
+    private void behaveOnNull(final ConnectRecord<R> r) {
+        switch (tombstoneHandlerConfig.getBehavior()) {
+            case FAIL:
+                throw new DataException("Record value or schema value can't be null");
+            case IGNORE:
+                LOGGER.debug("Record value or schema value can't be null");
+                break;
+            case WARN:
+                LOGGER.warn("Record value or schema value can't be null");
+                break;
+            default:
+                //skip it
```

Let's fail explicitly with an exception here to safeguard from potential bugs when a new behavior is added.